spring-integration-file/src/main/java/org/springframework/integration/file/filters/CompositeFileListFilter.java

```diff
@@ -53,13 +54,16 @@
 
 	private Consumer<F> discardCallback;
 
+	private boolean allSupportAccept = true;
+
 
 	public CompositeFileListFilter() {
 		this.fileFilters = new LinkedHashSet<>();
 	}
 
 	public CompositeFileListFilter(Collection<? extends FileListFilter<F>> fileFilters) {
 		this.fileFilters = new LinkedHashSet<>(fileFilters);
+		fileFilters.stream().allMatch(f -> f.supportsSingleFileFiltering());
```

Comment:
Are you missing `this.allSupportAccept &=` here?
What is the point of `allMatch()` if we don't use its result?

Reply:
Doh.

Reply:
Doh - that's exactly why I said more test coverage is needed 😄